### ISSUE TITLE
After upgrade from 2.20.27 to 2.22.4, when using npm run develop -- --https  saw ERR_SSL_PROTOCOL_ERROR in browser

### ISSUE THREAD

**Issue by frannnnk** [gatsbyjs/gatsby#24301]
## Description

After upgrade to 2.22.4, when using 

`npm run develop -- --https` 

to start the dev server, browser unable to open the page and show 

ERR_SSL_PROTOCOL_ERROR

### Steps to reproduce

- Perform a `yarn upgrade gatsby` 
- Run `npm run develop -- --https`  

Go to browser https://localhost:8000 

### Expected result

https website can be displayed

### Actual result

Browser show ERR_SSL_PROTOCOL_ERROR 

### Environment

  System:
    OS: macOS 10.15.4
    CPU: (8) x64 Intel(R) Core(TM) i7-1068NG7 CPU @ 2.30GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.16.3 - /usr/local/opt/node@12/bin/node
    Yarn: 1.21.0 - /usr/local/bin/yarn
    npm: 6.14.4 - /usr/local/opt/node@12/bin/npm
  Languages:
    Python: 2.7.15 - /usr/local/bin/python
  Browsers:
    Chrome: 81.0.4044.138
    Edge: 81.0.416.77
    Firefox: 68.0.1
    Safari: 13.1
  npmPackages:
    gatsby: ^2.20.0 => 2.22.4 
    gatsby-image: ^2.2.34 => 2.3.4 
    gatsby-plugin-react-helmet: ^3.1.16 => 3.2.4 
    gatsby-plugin-sass: ^2.1.24 => 2.2.3 
    gatsby-source-datocms: ^2.2.0 => 2.2.0 
    gatsby-source-filesystem: ^2.1.40 => 2.2.4 
    gatsby-transformer-remark: ^2.6.39 => 2.7.3 
  npmGlobalPackages:
    gatsby-cli: 2.12.21

**Comment by frannnnk** [comment]
It feels like connecting to a http server using https. but if I try to access http://localhost:8000 then the browser keep waiting and no response returned.

**Comment by hendra-go** [comment]
I believe this bug started from gatsby v2.22.0 after #22759 

after Develop with proxy server added.

Probably related to "proxy server with self signed certificate" ?

**Comment by JustFly1984** [comment]
@frannnnk  I confirm an issue, after upgrading from `2.21.40` to `2.22.6` `gatsby develop -S` stopped working. reverting back helped.

**Comment by mxstbr** [comment]
Closing as a duplicate of #24272, please track that issue!
